### ISSUE TITLE
Seek error when using latest version

### ISSUE THREAD

**Issue by rodcarroll** [fastavro/fastavro#217]
`File \"fastavro/_read.pyx\", line 597, in _iter_avro_records\n  File \"fastavro/_read.pyx\", line 610, in _iter_avro_blocks\nio.UnsupportedOperation: seek"`

I will try to add some more context, but I believe I am trying to read the file from a zip entry file object, which worked before version 0.19.7.

**Comment by scottbelden** [comment]
Thanks for the report. I have an idea of where the issue might be, but if you can provide a simple script to reproduce it or a little more information on how you are creating and iterating over the zip object then that would help.

**Comment by rodcarroll** [comment]
Here is some sample code that reproduces the issue:

**Comment by scottbelden** [comment]
I've been playing around with this some and added a test here: https://github.com/fastavro/fastavro/pull/218/

As you can see from the [travis output](https://travis-ci.org/fastavro/fastavro/builds/392863364), this works on python 3.7 but fails on all other versions. This is because the ability to call `seek` and `tell` were just recently added: https://bugs.python.org/issue22908.

I don't want to revert our usage of `tell` that was added recently, but as a workaround you should be able to do the following:

```python
import fastavro
from zipfile import ZipFile
from io import BytesIO


with ZipFile('weather.avro.zip', 'r') as fo:
    with fo.open('weather.avro', 'r') as zo:
        reader = fastavro.reader(BytesIO(zo.read()))  # <--- wrap zo
        for data in reader:
            print(data)
```

**Comment by rodcarroll** [comment]
Thanks for looking into this.  Part of the reason for reading directly from the zip file is to prevent the need to unzip large files on disk, so loading them into memory will not work unfortunately.

**Comment by scottbelden** [comment]
Understandable. As mentioned, this is fixed in python 3.7 so when you are able to upgrade to that then it should work. But for the time being you'll probably have to use the older version.
